### PR TITLE
Correct typo in "tools.go" for mc install on arm64

### LINKED ISSUE #410: `arkade get mc` on arm64 device fails
## Expected Behaviour

`arkade get mc` fails on an arm64 device.

## Current Behaviour

```
(base) emv@ny-c1-large-arm-01:~/src/github.com/alexellis/arkade$ arkade get mc
Downloading mc
{$arch = "arm64"}}

                        https://dl.min.io/client/mc/release/linux-aarch64/mc
Error: check with the vendor whether this tool is available for your system: parse "{$arch = \"arm64\"}}\n\t\t\t\n\t\t\thttps://dl.min.io/client/mc/release/linux-aarch64/mc": net/url: invalid control character in URL

```
## Possible Solution

Fix the tools.go file to correct the syntax.

## Steps to Reproduce (for bugs)

1. Get an arm64 device
2. `arkade get mc`

## Context

Working my way through the apps catalog for arm64.

## Your Environment

```
% arkade version
Version: 0.7.15
Git Commit: 426546ce181648f68380e06c572ae529f5cd6eba
```

## Are you a GitHub Sponsor (Yes/No?)

Check at: https://github.com/sponsors/alexellis
- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by vielmetti**
Closes #410

## Description

Typo fix in tools.go to allow correct installation of mc on arm64.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Open issue describing the problem is #410 

## How Has This Been Tested?

Not exhaustively tested, but by inspection the change is localized and corrects a typo.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by nitishkumar71** [comment]
@vielmetti Thanks for PR. It looks ok to me. Can you sign off the PR and push it again using `git commit --amend --signoff` and then `git push --force` on the same branch.

**Comment by alexellis** [comment]
We will need the unit test updating for this too. See how I did it for helm in a recent comment.

@vielmetti if you'd rather we take this over, just let us know.

**Comment by vielmetti** [comment]
@alexellis if you can take it over that would be great - if it looks like there's more than one place to fix things you'll find it far faster than I will.

**Comment by alexellis** [comment]
@nitishkumar71

**Comment by nitishkumar71** [comment]
Sure.

**Comment by shuuji3** [comment]
Hello🙂

I tried to find similar typos by searching regex (`[^{]\{\$`) on the repository wise and found the following typos:

```patch
diff --git a/pkg/get/tools.go b/pkg/get/tools.go
--- a/pkg/get/tools.go	(revision e14a995d20dd433368903a33f8ee7ddc742ee129)
+++ b/pkg/get/tools.go	(date 1621953682201)
@@ -867,7 +867,7 @@
 			{{- else if eq .Arch "armv7l" -}}
 			{{$arch = "arm"}}
 			{{- else if eq .Arch "aarch64" -}}
-			{$arch = "arm64"}}
+			{{$arch = "arm64"}}
 			{{- end -}}
 			{{$osStr := ""}}
 			{{ if HasPrefix .OS "ming" -}}
@@ -894,11 +894,11 @@
 			{{ if eq .Arch "x86_64" -}}
 			{{$arch = "amd64"}}
 			{{- else if eq .Arch "armv6l" -}}
-			{$arch = "arm6"}}
+			{{$arch = "arm6"}}
 			{{- else if eq .Arch "armv7l" -}}
-			{$arch = "arm7"}}
+			{{$arch = "arm7"}}
 			{{- else if eq .Arch "aarch64" -}}
-			{$arch = "arm64"}}
+			{{$arch = "arm64"}}
 			{{- end -}}
 			{{$osStr := ""}}
 			{{ if HasPrefix .OS "ming" -}}
@@ -980,7 +980,7 @@
 				{{- else if eq .Arch "armv7l" -}}
 				{{$arch = "armv7"}}
 				{{- else if eq .Arch "aarch64" -}}
-				{$arch = "arm64"}}
+				{{$arch = "arm64"}}
 				{{- end -}}
 
 				{{$versionString:=(printf "%s-%s" .OS $arch)}}
```

I hope this would help you.

File: pkg/get/tools.go [hunk 1]
```diff
@@ -861,7 +861,7 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
 			{{- else if eq .Arch "armv7l" -}}
 			{{$arch = "arm"}}
 			{{- else if eq .Arch "aarch64" -}}
-			{$arch = "arm64"}}
+			{{$arch = "arm64"}}
```
Comment (alexellis) [inline pkg/get/tools.go:864]: The test case was broken for this scenario saying "arm64" instead of "aarch64"

Comment (alexellis) [reply pkg/get/tools.go:864]: I also found nats had the same issue.

**Comment by alexellis** [comment]
@shuuji3 thanks I just saw your note.

Do you have a command line that we can run and add to CI to pick this up in the future?

@vielmetti I've fixed this myself since this was still WIP.

Alex

**Comment by shuuji3** [comment]
@alexellis In this time, I just searched the regex in my editor so there is no simple command on my side for now. But I think it would be nice if there is a test to check the validity of both `BinaryTemplate` and `URLTemplate` values of `tools`.

**Pull request closed without merging**